kitsune/wiki/views.py

```diff
@@ -1634,7 +1612,17 @@ def recent_revisions(request):
 def what_links_here(request, document_slug):
     """List all documents that link to a document."""
     locale = request.GET.get("locale", request.LANGUAGE_CODE)
-    doc = get_visible_document_or_404(request.user, locale=locale, slug=document_slug)
+
+    doc, fallback_code = get_visible_document_or_fallback_or_404(
+        request.user, locale, document_slug
+    )
+
+    if fallback_code is FallbackCodes.TRANSLATION_AT_DIFFERENT_SLUG:
```

Comment:
This pattern appears elsewhere too. I wonder if it makes sense to have a helper function here which will accept a translation as an arg and redirect or raise the appropriate HTTP errors